samza-core/src/main/scala/org/apache/samza/system/chooser/BootstrappingChooser.scala

```diff
@@ -266,9 +266,17 @@ class BootstrappingChooser(
 
     trace("Check %s offset %s against %s for %s." format (offsetType, offset, offsetToCheck, systemStreamPartition))
 
-    // The SSP is no longer lagging if the envelope's offset equals the
+    // Let's compare offset of the chosen message with offsetToCheck.
+    val comparatorResult: Integer = if (offset == null || offsetToCheck == null) {
+      -1
+    } else {
+      val systemAdmin = systemAdmins.getSystemAdmin(systemStreamPartition.getSystem)
+      systemAdmin.offsetComparator(offset, offsetToCheck)
+    }
+
+    // The SSP is no longer lagging if the envelope's offset is greater than or equal to the
     // latest offset.
-    if (offset != null && offset.equals(offsetToCheck)) {
+    if (comparatorResult != null && comparatorResult >= 0) {
```

Comment:
In most Java compilers, `comparatorResult == k` , will get compiled to `comparatorResult.compareTo(k) == 0` due to auto-boxing of types.

can make it explicit on need the for the null check - by using `comparatorResult.intValue() >= 0` instead.

Reply:
Just curious, can you explain what the intValue achieves? It would be an Integer comparision either way, right? Trying to avoid auto-boxing?